### PR TITLE
backports: for release v1.5.2

### PR DESCRIPTION
**Pull request by smira**
PRs backported:

- #7659 
- #7666 
- #7683
- #7692 
- #7702 
- #7704 
- #7705 
- #7706 
- #7707 
- Linux 6.1.51
- Go 1.20.8

### CODE REVIEW

File: internal/app/machined/pkg/adapters/cluster/identity_test.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func TestIdentityGenerate(t *testing.T) {
 	length := len(spec1.NodeID)
 
 	assert.GreaterOrEqual(t, length, 43)
-	assert.LessOrEqual(t, length, 44)
+	assert.LessOrEqual(t, length, 45)
```
Comment (frezbo) [inline internal/app/machined/pkg/adapters/cluster/identity_test.go:28]: what was this change?

Comment (smira) [reply internal/app/machined/pkg/adapters/cluster/identity_test.go:28]: just fixing flaky tests, this encoding has variable length in characters

**Pull request merged**